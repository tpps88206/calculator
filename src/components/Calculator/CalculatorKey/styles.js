const styles = theme => ({
  operatorKey: {
    color: 'white',
    fontSize: '3em',
    backgroundColor: '#3091FD',
  },
  digitKey: {
    color: 'white',
    fontSize: '3em',
    backgroundColor: '#333333',
  },
  functionKey: {
    fontSize: '2.5em',
    backgroundColor: '#AFAFAF',
    '& .MuiIcon-root': {
      display: 'flex',
    },
  },
  calculatorKey: {
    borderRadius: '50%',
    userSelect: 'none',
    cursor: 'pointer',
    outline: 'none',
    webkitTapHighlightColor: 'rgba(0,0,0,0)',
    '&:active': {
      boxShadow: 'inset 0px 0px 80px 0px rgba(0,0,0,0.25)',
    },
    '& .MuiIcon-root': {
      width: 'max-content',
      height: 'max-content',
      margin: 'auto',
    },
  },
});

export default styles;
