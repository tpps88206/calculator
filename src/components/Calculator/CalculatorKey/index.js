import React from 'react';
import PointTarget from 'react-point';

import classNames from 'classnames';

import { makeStyles } from '@material-ui/styles';

import styles from './styles';

const useStyles = makeStyles(styles);

const CalculatorKey = ({ onPress, variant, className, ...props }) => {
  const classes = useStyles();

  return (
    <PointTarget onPoint={onPress}>
      <button
        className={classNames(
          className,
          classes.calculatorKey,
          {
            [classes.operatorKey]: variant === 'operator-key',
            [classes.digitKey]: variant === 'digit-key',
            [classes.functionKey]: variant === 'function-key',
          },
          'd-block',
          'w-100',
          'h-100',
        )}
        {...props}
      />
    </PointTarget>
  );
};

export default CalculatorKey;
