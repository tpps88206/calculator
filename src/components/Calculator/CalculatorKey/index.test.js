import React from 'react';

import { render } from '../../../utils/test-utils';
import CalculatorKey from './index';

it('renders CalculatorKey without crashing', () => {
  const div = document.createElement('div');
  render(<CalculatorKey />, div);
});

it('renders CalculatorKey with snapshot', () => {
  const { asFragment } = render(<CalculatorKey />);
  expect(asFragment()).toMatchSnapshot();
});
