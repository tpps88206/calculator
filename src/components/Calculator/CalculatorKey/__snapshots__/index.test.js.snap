// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CalculatorKey with snapshot 1`] = `
<DocumentFragment>
  <button
    class="makeStyles-calculatorKey-8 d-block w-100 h-100"
  />
</DocumentFragment>
`;
